### ISSUE TITLE
trainer.py: gradient clipping at 10.0 may be too permissive for transformer architecture

### ISSUE THREAD

**Issue by ArgonZapan** [ArgonZapan/DQN-transformer#7]
## Performance / Training Stability Issue

In `python/training/trainer.py` at line 157, gradient clipping is set to a relatively high value:

```python
torch.nn.utils.clip_grad_norm_(self.main_network.parameters(), 10.0)
```

For transformer-based architectures (which this DQN uses), large transformer models and deep networks are prone to gradient explosions even at lower clipping thresholds. Standard values for transformer training are typically **1.0 to 5.0**, not 10.0.

A clipping value of 10.0 may allow large gradient spikes to pass through, potentially destabilizing training, especially as the network depth increases with additional transformer blocks.

## Suggested Fix

Test with more conservative clipping values:
- `clip_grad_norm_(..., 1.0)` — strictest
- `clip_grad_norm_(..., 2.0)` — moderate, standard for transformers
- `clip_grad_norm_(..., 5.0)` — moderate-high

Monitor gradient norms during training (add logging of `grad_norm` before clipping) and choose a threshold that captures >95% of gradient updates but still clips clear outliers.

## Labels

enhancement, performance

**Comment by ArgonZapan** [comment]
Fixed in commit 018aa93: gradient clipping lowered from 10.0 to 1.0, which is standard for transformer architectures.
